source/patterns/@aws-solutions-constructs/aws-apigatewayv2websocket-sqs/test/apigatewayv2websocket-sqs.test.ts

```diff
@@ -0,0 +1,339 @@
+      CredentialsArn: { "Fn::GetAtt": [Match.anyValue(), "Arn"] },
+      IntegrationMethod: "POST",
+      IntegrationType: "AWS",
+      IntegrationUri: { "Fn::Join": Match.anyValue() },
+      PassthroughBehavior: "NEVER",
+      RequestParameters: { "integration.request.header.Content-Type": "'application/x-www-form-urlencoded'" },
+    });
+
+    template.hasResourceProperties("AWS::ApiGatewayV2::Route", {
+      RouteKey: "$default",
+      AuthorizationType: "NONE",
+      Target: {
+        "Fn::Join": ["", ["integrations/", Match.anyValue()]],
+      },
+    });
+  });
+});
+
+describe("When instantiating the construct with existing WebSocketApi", () => {
+  let template: Template;
+
+  beforeAll(() => {
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, "TestStack");
+    const mockConnectLambda = new lambda.Function(stack, "mockConnectFunction", {
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+      runtime: COMMERCIAL_REGION_LAMBDA_NODE_RUNTIME,
+      handler: "connect.handler",
+    });
+
+    const mockDisconnectLambda = new lambda.Function(stack, "mockDisconnectFunction", {
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+      runtime: COMMERCIAL_REGION_LAMBDA_NODE_RUNTIME,
+      handler: "disconnect.handler",
+    });
+
+    new ApiGatewayV2WebSocketToSqs(stack, "ApiGatewayV2WebSocketToSqs", {
+      existingWebSocketApi: new apigwv2.WebSocketApi(stack, "TestWebSocket", {
+        apiName: "TestWebSocket",
+        description: "Test WebSocket",
+        connectRouteOptions: {
+          integration: new WebSocketLambdaIntegration("ConnectIntegration", mockConnectLambda),
+          authorizer: new WebSocketIamAuthorizer(),
+        },
+        disconnectRouteOptions: {
+          integration: new WebSocketLambdaIntegration("DisconnectIntegration", mockDisconnectLambda),
+        },
+      }),
+    });
+    template = Template.fromStack(stack);
+  });
+
+  it("should not create a new websocket but use existing one", () => {
+    template.resourceCountIs("AWS::ApiGatewayV2::Api", 1);
+    template.hasResourceProperties("AWS::ApiGatewayV2::Api", {
+      Name: "TestWebSocket",
+      Description: "Test WebSocket",
+    });
+  });
+
+  it("should have 2 routes configured", () => {
+    template.resourceCountIs("AWS::ApiGatewayV2::Route", 2);
+
+    template.hasResourceProperties("AWS::ApiGatewayV2::Route", {
+      RouteKey: "$disconnect",
+      AuthorizationType: "NONE",
+      Target: {
+        "Fn::Join": ["", ["integrations/", Match.anyValue()]],
+      },
+    });
+
+    template.hasResourceProperties("AWS::ApiGatewayV2::Route", {
+      RouteKey: "$connect",
+      AuthorizationType: "AWS_IAM",
+      Target: {
+        "Fn::Join": ["", ["integrations/", Match.anyValue()]],
+      },
+    });
+  });
+});
+
+describe("When an existing instance of WebSocketApi and WebSocketApiProps both are provided", () => {
```

Comment:
mentioned above, but also a test when neither are provided would be good.

Reply:
unit test added